### PR TITLE
chore(docs): review messaging docs

### PR DESCRIPTION
**Pull request by adrianbenavides**
This PR includes a bunch of typo fixes as well as some rephrasings of some sentences that seemed confusing to me.

The [first commit](https://github.com/ockam-network/ockam/pull/2660/commits/74c021f8fd87fb0cb8e4ed5e0caa5c902c8c4dd3) includes the fixes, the second commit applies the formatting, so I'd recommend checking the first commit in isolation to see the fixes more clearly.

### CODE REVIEW

**Review by hairyhum — COMMENTED** [review]
Line breaks in some places are intended to deliver logical statements. I'm not sure it would be right to just remove them. Maybe we can use some other way to highlight that?

File: documentation/architecture/messaging/Accessibility.md [hunk 1]
```diff
@@ -152,46 +145,39 @@ Backtracing is different for route-based, static and metadata based forwarders.
...
+- if the forwarder `F` receives the message from `A` with return route `F->A`, it should forward the message with return
+  route `[0#F'] ; F->A`
+- if the forwarder `F` receives the message from `B` with return route `F->B`, it should forward the message with return
+  route `[0#F] ; F->B`
 
 This makes worker `A` and `B` mutually accessible via routes `A->F ; F'->B` and backtrace `B->F' ; F->A`
 
 <img src="./images/proxy.jpg" width="100%">
 
-Such forwarding worker is called **proxy worker**
```
Comment (hairyhum) [inline documentation/architecture/messaging/Accessibility.md:172]: Let's keep the separate lines here

File: documentation/architecture/messaging/Reliability.md [hunk 2]
```diff
@@ -1,125 +1,119 @@
...
-**A difference between pipe reliability and delivery reliability is that pipe does not take into account delivery between pipe workers and application workers**
+This allows us to design the pipes as building blocks to achieve reliable delivery, **with the caveat of having actual endpoints and pipe
+workers always running and not having errors*.
 
+**A difference between pipe reliability and delivery reliability is that pipe does not take into account delivery between pipe workers and
+application workers**.
 
 ## Improving reliability
 
-Retries can be used to improve delivery reliability.
```
Comment (hairyhum) [inline documentation/architecture/messaging/Reliability.md:94]: Let's use a list here 
```
- Given
- If
- Then
```

File: documentation/architecture/messaging/Reliability.md [hunk 3]
```diff
@@ -1,125 +1,119 @@
...
+then `R∞->1`.
 
-If we retry delivery on missing message, it becomes `R1 = R + (1- R) * R = 0.75` on first retry, `R2 = 0.875` on second, etc.
+Retries **decrease the uniqueness rate**. Given `DM(A->B) = {0.5, 1}`, if we enable retries for this delivery
+then `DMr∞(A->B) = {->1, ->0.5}`, where reliability increased at the expense of reducing the uniqueness rate.
 
-If retries `-> ∞`, then `R∞->1`
-
-Retries decrease uniqueness rate
-Given `DM(A->B) = {0.5, 1}`
```
Comment (hairyhum) [inline documentation/architecture/messaging/Reliability.md:103]: Let's use a list here 
```
- Given
- If
- Then
```

File: documentation/architecture/messaging/Reliability.md [hunk 4]
```diff
@@ -131,70 +125,64 @@ This is called **deduplication**, when we ignore some received messages based on
...
 ## Reliable message delivery in presence of errors in the pipeline
 
-Previously we assumed that workers are always alive and have no errors, in that case given worker forwards all messages exactly-once it's internal delivery mode will be exactly-once.
+Previously we assumed that workers are always alive and have no errors. In that case, given that a worker forwards all messages
+exactly-once, its internal delivery mode will be also exactly-once.
 
-If it's not so, then we should consider reliability of delivery from worker receiving a message to it sending this message.
+Otherwise, we should consider reliability of delivery from worker receiving a message to it sending this message.
 
 If we have a pipeline of routes `A->X; X->B`, in which `R(A->X) -> 1` and `R(X->B) -> 1`
```
Comment (hairyhum) [inline documentation/architecture/messaging/Reliability.md:140]: Let's use a list here 
```
- If
- and
- Then
```

File: documentation/architecture/messaging/Sessions.md [hunk 5]
```diff
@@ -8,100 +8,94 @@ Basic session concepts are described here: [Sessions discussion thread](https://
...
+Reply route may lead to some discovery worker, which will save the spawned worker route and make it available to other workers.
 
 <img src="./images/spawner_separate_discovery.jpg" width="80%">
 
-Spawner worker may serve as a discovery worker if reply route leads back to the spawner
+Spawner worker may serve as a discovery worker if reply route leads back to the spawner.
 
 <img src="./images/spawner_discovery.jpg" width="50%">
 
-If "create" message route is backtraced, reply route can be a backtrace to the original message.
```
Comment (hairyhum) [inline documentation/architecture/messaging/Sessions.md:76]: Let's keep lines separate here

**Review by hairyhum — CHANGES_REQUESTED** [review]

**Comment by adrianbenavides** [comment]
Sorry for the delay @hairyhum, I just kept forgetting about this PR. I've applied all the changes you requested, so it should be ready now!

**Review by hairyhum — APPROVED** [review]

**Comment by hairyhum** [comment]
@adrianbenavides please rebase and squash

**Comment by adrianbenavides** [comment]
> @adrianbenavides please rebase and squash

ready!

**Review by hairyhum — APPROVED** [review]

**Pull request merged**